### PR TITLE
Create Scenario

### LINKED ISSUE #1125: Create Scenario
**Description**

Compass User should be able to create new scenario from scenario overview page

**AC**
 - User should specify scenario name
 - User may specify applications and runtimes

Screen Shot 2019-08-01 at 11 21 41

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Add modal for creating scenarios
- Refactor list used in assigning scenarios / applications / runtimes
- Fix tiny _add application_ bug

**Related issue(s)**
Resolves #1125

### CODE REVIEW

File: compass/src/components/Scenarios/CreateScenario/AssignEntityList.component.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import { Menu } from 'fundamental-react';
+import './style.scss';
+
+AssignEntityList.propTypes = {
+  placeholder: PropTypes.string,
+  entitiesQuery: PropTypes.object.isRequired,
+  updateEntities: PropTypes.func.isRequired,
+};
+
+export default function AssignEntityList({
```
Comment (dariadomagala) [inline compass/src/components/Scenarios/CreateScenario/AssignEntityList.component.js]: Do we need two separate components that do basically the same thing? (this one + assign scenario)

File: compass/src/components/Scenarios/CreateScenario/CreateScenarioModal/CreateScenarioModal.component.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    } else {
+      this.setState({ nameError: '' });
+    }
+  };
+
+  updateApplications = (assignedApplications, _) => {
+    this.setState({ applicationsToAssign: assignedApplications });
+  };
+
+  updateRuntimes = (assignedRuntimes, _) => {
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply compass/src/components/Scenarios/CreateScenario/CreateScenarioModal/CreateScenarioModal.component.js]: The same question to the function above.

File: compass/src/components/Shared/MultiChoiceList/MultiChoiceList.component.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  currentlyNonSelectedItems,
+  updateItems,
+  notSelectedMessage,
+  noEntitiesAvailableMessage,
+  displayPropertySelector,
+}) {
+  const [selectedItems, setSelectedItems] = React.useState(
+    currentlySelectedItems,
+  );
+  const [nonSelectedItems, setNonselectedItems] = React.useState(
```
Comment (parostatkiem) [inline compass/src/components/Shared/MultiChoiceList/MultiChoiceList.component.js]: `setNonselectedItems` -> `setNonSelectedItems`

File: compass/src/components/Shared/MultiChoiceList/MultiChoiceList.component.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  }
+
+  function updateLists(newSelectedItems, newNonSelectedItems) {
+    setSelectedItems(newSelectedItems);
+    setNonselectedItems(newNonSelectedItems);
+
+    updateItems(newSelectedItems, newNonSelectedItems);
+  }
+
+  function createAssignedEntitiesList() {
```
Comment (parostatkiem) [inline compass/src/components/Shared/MultiChoiceList/MultiChoiceList.component.js]: I suggest the following
```javascript
if(!selectedItems.length) return;
```
then defining `selectedEntitiesList` won't be needed anymore.

File: compass/src/components/Shared/MultiChoiceList/MultiChoiceList.component.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      </Dropdown>
+    );
+
+    const noEntitiesMessage = (
+      <span className="fd-has-font-style-italic">
+        {noEntitiesAvailableMessage}
+      </span>
+    );
+
+    return nonSelectedItems.length ? dropdown : noEntitiesMessage;
```
Comment (parostatkiem) [inline compass/src/components/Shared/MultiChoiceList/MultiChoiceList.component.js]: The same as above - you can return `noEntitiesMessage` at the beginning.

File: compass/src/components/Scenarios/CreateScenario/CreateScenarioModal/CreateScenarioModal.component.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+      const labels = runtime.labels.scenarios || [];
+      return setRuntimeScenarios(runtime.id, [...labels, scenarioName]);
+    });
+
+    const result = await Promise.allSettled([
+      ...applicationUpdates,
+      ...runtimeUpdates,
+    ]);
+
+    const rejected = result.filter(r => r.status === 'rejected');
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply compass/src/components/Scenarios/CreateScenario/CreateScenarioModal/CreateScenarioModal.component.js]: You're right, but anyway the integer is used few lines below.

**Pull request merged**